File: java_implementation/src/main/java/com/tambapps/web/page_scrapping/parameters/validator/NaturalInteger.java
package com.tambapps.web.page_scrapping.parameters.validator;

import com.beust.jcommander.IValueValidator;
import com.beust.jcommander.ParameterException;

public class NaturalInteger implements IValueValidator<Integer> {

  @Override
  public void validate(String name, Integer n) throws ParameterException {
    if (n <= 0) {
      throw new ParameterException("Parameter " + name + " should be natural (found " + n + ")");
    }
  }
}

File: kotlin_implementation/src/main/kotlin/com/tambapps/kotlin/parameters/validator/NaturalInteger.kt
package com.tambapps.kotlin.parameters.validator

import com.beust.jcommander.IValueValidator
import com.beust.jcommander.ParameterException

class NaturalInteger: IValueValidator<Int> {
    override fun validate(name: String, n: Int) {
        if (n <= 0) {
            throw ParameterException("Parameter $name should be natural (found $n)")
        }
    }
}